### PR TITLE
Readiness & liveness probes correct port

### LINKED ISSUE #11461: Chart generator puts hard-coded port 80 in resulting deployment.yaml
The `deployment.yaml` template created from `helm create ...` works but port 80 is hard-coded. This impacts deployment of apps that use non-80 port internally (e.g. React apps on 3000). Changing the port used in `values.yaml` prevents the pod from launching because liveness and ready endpoints are incorrect.

Based on my very limited knowledge of Helm/Go, it appears [line 315 of pkg/chartutil/create.go](https://github.com/helm/helm/blob/main/pkg/chartutil/create.go#L315) needs to change 80 to `{{ .Values.service.port }}`.

Output of `helm version`:  
```
version.BuildInfo{Version:"v3.10.1", GitCommit:"9f88ccb6aee40b9a0535fcc7efea6055e1ef72c9", GitTreeState:"clean", GoVersion:"go1.18.7"}
```

Output of `kubectl version`:
```
clientVersion:
  buildDate: "2022-09-21T14:33:49Z"
  compiler: gc
  gitCommit: 5835544ca568b757a8ecae5c153f317e5736700e
  gitTreeState: clean
  gitVersion: v1.25.2
  goVersion: go1.19.1
  major: "1"
  minor: "25"
  platform: linux/amd64
kustomizeVersion: v4.5.7
serverVersion:
  buildDate: "2022-09-21T14:27:13Z"
  compiler: gc
  gitCommit: 5835544ca568b757a8ecae5c153f317e5736700e
  gitTreeState: clean
  gitVersion: v1.25.2
  goVersion: go1.19.1
  major: "1"
  minor: "25"
  platform: linux/amd64
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): Minikube

### PR DESCRIPTION
**Pull request by p33t**
Port number now comes from values.yaml

closes #11461 

**What this PR does / why we need it**: Fixes port number in generated `deployment.yaml` (during `helm create ...`)

**Special notes for your reviewer**: This works but might not be the right approach

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by joejulian — APPROVED** [review]
Nice catch. I approve, though I'm not a core maintainer. This needs a review from one of them before it can be merged.

**Review by cndoit18 — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]

**Pull request merged**
